platform/engineering/design-docs/2022-03-11-schedule-pending-appt.md

```diff
@@ -0,0 +1,170 @@
+# Scheduling Pending Appointments
+
+**Author(s):** Andrew Herzberg  
+**Last Updated:** March 11,2022  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** _Person A_ \[ \], _Person B_ \[ \], ...  
+
+## Overview
+
+### Objective
+The goal is to add the feature to the mobile app to schedule pending appointments. This functionality is currently available in `vets-website` so much of the logic can be utilized in the mobile app.
+
+The intended audience is for the engineering team.
+
+### Background
+Currently, the `vets-website` scheduling feature is as follows:
+
+![XL9BReCm4DrpYhb0Ue0igaIehRfgrQZT4tYGQOrDFKCaRbz3BOXYb6mGiZtly-RjH0klVMkoJu6dpICTuC_mMyfTs_QMTQ2yy11wCjpmaSruye7V7g94cNKACgXrIs_M35IUKFu8rSw7NoUmf4C7SZL9mEKIsJy-RG9LeBnB99spWkmq4RbnSX8QnaxE2OI-zXJ8yfoudhv2c00C4alFHJd2CeHURP3O5ihEhhgP3-jbflQk](https://user-images.githubusercontent.com/1910447/158439763-fd97627e-1bb7-4053-bb30-86436b64a2a4.png)
+
+- Pick type of care to schedule an appointment for
```

Comment:
Do we know if this is specific to the user? We know there are some eligibility requirements around "user is registered at facility and does not have more than two outstanding requests at facility". Do these endpoints do those checks? I assume they do, but if they don't we could set up an overnight cron job that collects this info for all facilities and caches it, which would substantially improve response time.

Reply:
The v0 version of this process (as documented here: https://github.com/department-of-veterans-affairs/va.gov-team/pull/35452/files) also fetched the past two years of appointments, and I speculated that it was how we fetched the available facilities. I believe two years is the cut off we've heard for facility eligibility, so that would be consistent. I'm really not sure why the web app fetches past appointments here, but it would be kinda crazy to have the front end fetch past appointments and have its own logic to determine eligibility. That logic should definitely be in the back end.